static/app/components/group/externalIssuesList/streamlinedExternalIssueList.tsx

```diff
@@ -112,143 +104,152 @@ export function StreamlinedExternalIssueList({
                 }
                 isHoverable
               >
-                <LinkedIssue
+                <LinkButton
                   href={linkedIssue.url}
                   external
                   size="zero"
                   icon={linkedIssue.displayIcon}
                 >
                   <IssueActionName>{linkedIssue.displayName}</IssueActionName>
-                </LinkedIssue>
+                </LinkButton>
               </Tooltip>
             </ErrorBoundary>
           ))}
         </IssueActionWrapper>
       )}
-      {integrations.length > 0 && (
-        <IssueActionWrapper>
-          {integrations.map(integration => {
-            const sharedButtonProps: ButtonProps = {
-              size: 'zero',
-              icon: integration.displayIcon,
-              children: <IssueActionName>{integration.displayName}</IssueActionName>,
-            };
+      <ExternalIssueMenu linkedIssues={linkedIssues} integrations={integrations} />
+    </Flex>
+  );
+}
 
+function ExternalIssueMenu(props: ReturnType<typeof useGroupExternalIssues>) {
+  const organization = useOrganization({allowNull: false});
+
+  return (
+    <Fragment>
+      <CompositeSelect
+        trigger={triggerProps => (
+          <Button {...triggerProps} size="zero" icon={<IconAdd />}>
+            {props.linkedIssues.length === 0 ? t('Add Linked Issue') : null}
+          </Button>
+        )}
+        // Required for submenu interactions
+        isDismissable={false}
+        menuTitle={t('Add Linked Issue')}
+        hideOptions={props.integrations.length === 0}
+        menuBody={props.integrations.length === 0 && <ExternalIssueMenuEmpty />}
+        menuFooter={props.integrations.length > 0 && <ExternalIssueManageLink />}
+      >
+        <CompositeSelect.Region
+          closeOnSelect={({value}) => {
+            const integration = props.integrations.find(({key}) => key === value);
+            if (!integration) {
+              return true;
+            }
```

Comment:
also for the future: the fact that we need to map over `props.integrations` again here to find a value (that certainly exists!) seems like a design flaw in `CompositeSelect`. It’s too bad that we aren’t generic on the whole option itself - only on the value.

If `options` weren’t `options: Array<SelectOption<Value>>` but rather fully generic over the Option itself: `options: Array<TSelectOption>` where `TSelectionOption extends SelectOption`, we could pass in the whole `props.integration` if we want and just extract `actions` from the param we get passed  in to `closeOnSelect` and `onChange`. At runtime, we already get the full option passed in.

It’s these kinds of things that would make working with our core components much easier.